docs/src/architecture/09_architecture_decisions/0006-cursor-pegination-cat-data-service.md

```diff
@@ -0,0 +1,118 @@
+---
+    title: 0006 Cursor-based pagination for Catalyst Data Service
+    adr:
+        author: Oleksandr Prokhorenko
+        created: 05-Feb-2024
+        status:  proposed
+        extends:
+    tags:
+        - api
+---
+
+## Context
+
+Our API currently lacks a pagination mechanism, leading to inefficiencies in data retrieval processes,
+especially when dealing with large datasets.
+This absence has resulted in longer load times and
+a strained user experience, prompting the need for a scalable solution to manage data access and navigation effectively.
+
+## Assumptions
+
+* The assumption that cursor-based pagination will be universally supported and understood by all clients consuming our API.
+* Assuming our database and backend infrastructure can efficiently support cursor-based operations without
+ significant refactoring or performance degradation.
+* The expectation that data growth will continue at its current pace or accelerate,
+necessitating a robust solution.
+
+## Decision
+
+We are proposing to introduce a cursor-based pagination system to our API.
+This system will rely on ans enabling clients to efficiently fetch data in chunks and
+understand when more data is available:
+
+* `limit` - This establishes a restriction on the quantity of objects to be returned,
+with a minimum of 1 and a maximum of 100.
+* `starting_after`- the cursor to start returning results from,
+* `ending_before`- the cursor to stop returning results at.
+
+The JSON response structure will include fields such as
+
+* `data` - containing the requested data,
```

Comment:
Should we say that data must always be deterministically sorted, such that `id` always return in the same order, regardless of the pagination?